idunn/blocks/opening_hour.py

```diff
@@ -82,38 +75,27 @@ def parse_time_block(cls, es_poi, lang, raw):
         )
         return None
 
-    poi_dt = UTC.localize(datetime.utcnow()).astimezone(poi_tz)
+    poi_dt = utc.localize(datetime.utcnow()).astimezone(poi_tz)
 
-    if oh.is_open(poi_dt.replace(tzinfo=None)):
+    if oh.is_open():
         status = cls.STATUSES[0]
     else:
         status = cls.STATUSES[1]
 
-    if cls.BLOCK_TYPE == OpeningHourBlock.BLOCK_TYPE and (raw == "24/7" or oh.is_24_7):
+    if cls.BLOCK_TYPE == OpeningHourBlock.BLOCK_TYPE and oh.is_24_7():
         return cls.init_class(status, None, None, oh, poi_dt, raw)
 
-    # The current version of the hoh lib doesn't allow to use the next_change() function
-    # with an offset aware datetime.
-    # This is why we replace the timezone info until this problem is fixed in the library.
-    try:
-        nt = oh.next_change(dt=poi_dt.replace(tzinfo=None))
-    except HOHError:
-        logger.info(
-            "HOHError: Failed to compute next transition for poi %s",
-            es_poi.get("id"),
-            exc_info=True,
-        )
+    next_transition = oh.next_change()
```

Comment:
`is_open` and `next_change` should be evaluated on the exact same datetime `poi_dt`, to avoid edge cases when this function would be called at the exact moment when the state is changing, and the output would then be consistent.

Improbable, but definitely possible.